### ISSUE TITLE
HiDPI Auto Scaling on Linux

### ISSUE THREAD

**Issue by cristatus** [JFormDesigner/FlatLaf#69]
Do not scale if `GDK_SCALE` environment variable is set. Also, do not auto scale if app is being run with [JetBrain-Runtime](https://confluence.jetbrains.com/display/JBR/JetBrains+Runtime) which has auto scaling enabled by default.

**Comment by DevCharly** [comment]
Thanks for pointing out. I'll have a look...

**Comment by cristatus** [comment]
I am not sure whether it is a correct fix or not but changing this line [LinuxFontPolicy.java#L174](https://github.com/JFormDesigner/FlatLaf/blob/master/flatlaf-core/src/main/java/com/formdev/flatlaf/LinuxFontPolicy.java#L174) as follows fixes the issue.

```java
if( dpi == -1 || UIScale.isSystemScalingEnabled() )
```

**Comment by DevCharly** [comment]
@cristatus I've fixed this issue in master branch. Could you please test it? Thx.

**Comment by cristatus** [comment]
@DevCharly I have just checked and it's working but fonts are smaller. I think because of dpi calculation.

Here I am attaching screenshots:

![shot-1](https://user-images.githubusercontent.com/282290/77655872-0acb5a80-6f99-11ea-8024-15d6bed38178.png)

![shot-2](https://user-images.githubusercontent.com/282290/77655882-1159d200-6f99-11ea-8797-581364d0b18b.png)

The first one is as per my fix and the next one is with master branch.

**Comment by cristatus** [comment]
I have just tried removing first 3 lines from `getGnomeFontScale()` and added this condition `if( dpi == -1 || isSystemScaling() )` as I mentioned earlier and issue is fixed.

**Comment by cristatus** [comment]
@DevCharly I can confirm it's working fine with the changes I proposed in my last comment.

Tested with jetbrain runtime and AdoptOpenJDK with and without `GDK_SCALE`.

**Comment by DevCharly** [comment]
@cristatus many thanks for testing.

I now realize the bug in my fix and the difference to your fix.

It should be:
~~~java
if( isSystemScaling() )
    return 96. / 72.;
~~~

I think it is better to return early from `getGnomeFontScale()` for the case that the desktop property "gnome.Xft/DPI" is not set.

BTW what Linux distro do you use? It looks very nice. Want try this distro...

**Comment by DevCharly** [comment]
The change is in master branch.

**Comment by cristatus** [comment]
@DevCharly thanks! I will give it a try soon. BTW, it's Arch Linux with Gnome 3.36.

**Comment by cristatus** [comment]
@DevCharly now it's working fine!

**Comment by martin-jamszolik** [comment]
Thank you guys for working this issue out.  Eager to see this fix land in the next version of Netbeans.
I am also a linux workstation user ( Fedora 31).  Thanks to this project, Java Swing looks great again. I can finally restore my hope in being able to use these kind of apps in 4K.

**Comment by DevCharly** [comment]
@martin-jamszolik thanks. I'll release a new FlatLaf version the next days and then submit a PR to NetBeans. If all goes well, the fix should land in NetBeans 12 beta 3.

**Comment by cristatus** [comment]
Thanks @DevCharly for the great work. Few edge cases are there but I am sure it will become more perfect over time. Do you have any plan to add shadow support for menu and other popups like tooltips?

**Comment by DevCharly** [comment]
fixed in 0.29

**Comment by DevCharly** [comment]
> Do you have any plan to add shadow support for menu and other popups like tooltips?

Not yet.

**Comment by DevCharly** [comment]
Fixed in NetBeans 12.0 beta 3, which is now available:
https://builds.apache.org/view/M-R/view/NetBeans/job/netbeans-TLP/job/netbeans/job/release120/12/artifact/dist/netbeans/

**Comment by DevCharly** [comment]
@cristatus drop shadows are now implemented in master branch and will be released in FlatLaf 0.35

**Comment by cristatus** [comment]
> @cristatus drop shadows are now implemented in master branch and will be released in FlatLaf 0.35

Cool! I will give it a try soon. Thanks!

**Comment by cristatus** [comment]
@DevCharly I just tried and found few issues. Looks like it's being tracked with #94, I will provide my feedback there.
